tests/main/interfaces-contacts-service/task.yaml

```diff
@@ -20,14 +20,25 @@ environment:
     XDG_DATA_HOME: $XDG/share
     XDG_CACHE_HOME: $XDG/cache
 
+debug: |
+    # debug for when the "rm -rf $XDG" in restore fails
+    systemctl status gvfs-metadata || true
+    systemctl --user status gvfs-metadata || true
+    ls -alR "$XDG/share/" || true
+    ps afx
+
 restore: |
     if [ -e dbus-launch.pid ]; then
         kill "$(cat dbus-launch.pid)"
     fi
 
+    # gvfsd-metadata locks the xdg/share/gvfs-metadata directory content
+    # producing an error when the xdg directory is removed. So we need
+    # to stop it:
+    if systemctl --user is-active gvfs-metadata; then
+        systemctl --user stop gvfs-metadata
```

Comment:
do we maybe need something like the following:

```bash
systemctl kill --kill-who=all gvfs-metadata
while ! (systemctl list-units --all gvfs-metadata | egrep -q '(dead|failed)')
do
  sleep 1;
done
```

I'm not familiar with gvfs-metadata but if it's getting stuck somewhere where even stop isn't working properly then maybe we need to go nuclear with `kill` and wait for it to become dead/failed.

Reply:
FWTW we kill -9 it below in restore.